drivers/gpio/hcsr04_driver.go

```diff
@@ -0,0 +1,166 @@
+package gpio
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"gobot.io/x/gobot"
+	"gobot.io/x/gobot/drivers/gpio"
+)
+
+const (
+	soundSpeed float32 = 343.0 // in [m/s]
+	// the device can measure 2cm .. 4m, this means sweep distances between 4cm and 8m
+	// this cause pulse durations between 0.12ms and  24ms (at 34.3 cm/ms, ~0.03 ms/cm, ~3ms/m)
+	measurementCycle time.Duration = 60 // 60ms between two measurements
+	MonitorUpdate    time.Duration = 100 * time.Millisecond
+)
+
+// HCSR04 instance
+type HCSR04 struct {
+	name                   string
+	connection             gobot.Adaptor
+	triggerPin             *gpio.DirectPinDriver
+	echoPin                *gpio.DirectPinDriver
+	mux                    sync.Mutex
+	Measure                float32 // The last measure
+	distanceMonitorControl chan int
+	distanceMonitorStarted bool
+	gobot.Commander
+}
+
+// NewHCSR04 creates a new HCSR04 instance
+func NewHCSR04(a gobot.Adaptor, triggerPinID string, echoPinID string) *HCSR04 {
+	return &HCSR04{
+		name:       gobot.DefaultName("HCSR04"),
+		triggerPin: gpio.NewDirectPinDriver(a, triggerPinID),
+		echoPin:    gpio.NewDirectPinDriver(a, echoPinID),
+		connection: a,
+		Commander:  gobot.NewCommander(),
+	}
+}
+
+// MeasureDistance measure the distance in front of sensor in meters
+// and returns the measure
+// MeasureDistance triggers a distance measurement by the sensor
+//
+// ! MeasureDistance is not design to work in a fast loop
+// For this specific usage, use StartDistanceMonitor associated with GetDistance Instead
+func (hcsr04 *HCSR04) MeasureDistance() (float32, error) {
+	hcsr04.mux.Lock()
+	defer hcsr04.mux.Unlock()
+	pulseDuration, err := hcsr04.measurePulse()
+	if err != nil {
+		return 0, err
+	}
+	hcsr04.Measure = pulseToDistance(pulseDuration)
+	return hcsr04.Measure, nil
+}
+
+func (hcsr04 *HCSR04) emitTrigger() {

```

Comment:
Please move all local functions at the end.

Reply:
Done.